include/bluetooth/att.h

```diff
@@ -59,10 +59,37 @@ extern "C" {
 #if defined(CONFIG_BT_TESTING)
 
 int bt_eatt_disconnect_one(struct bt_conn *conn);
-int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels);
-size_t bt_eatt_count(struct bt_conn *conn);
 
 #endif /* CONFIG_BT_TESTING */
+
+/** @brief Connect Enhanced ATT channels
+ *
+ * Sends a series of Credit Based Connection Requests to connect @p num_channels
+ * Enhanced ATT channels. The peer may have limited resources and fewer channels
+ * may be created.
+ *
+ * @param conn The connection to send the request on
+ * @param num_channels The number of Enhanced ATT beares to request.
+ * Must be in the range 1 - @kconfig{CONFIG_BT_EATT_MAX}, inclusive.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ * @retval -EINVAL if @p num_channels is not in the allowed range.
+ * @retval -EINVAL if @p conn is NULL.
+ * @retval -ENOMEM if less than @p num_channels are allocated.
+ * @retval 0 in case of success
+ */
+int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels);
+
+/** @brief Get number of EATT channels connected.
+ *
+ * @param conn The connection to get the number of EATT channels for.
+ *
+ * @return The number of EATT channels connected.
+ * @retval -EINVAL if @p conn is NULL.
+ * @retval -ENOTCONN @p conn is not connected.
+ */
+int bt_eatt_count(struct bt_conn *conn);
```

Comment:
I'd keep the original `size_t` return value and simply return `0` in the situations where you're now trying to return a negative error value.

Reply:
On that note, it would then also be nice to be consistent with the `num_channels` parameter to `bt_eatt_connect()`, i.e. I'd probably change that to `size_t` as well.

Reply:
> I'd keep the original `size_t` return value and simply return `0` in the situations where you're now trying to return a negative error value.

Not sure I agree with this, since there can be multiple reasons why it fails. 

One alternative that may be better is to use the `ssize_t` which is commonly used to return -1 in case of error. Then we have values -1 and 0 to work with for error or unsuccessful conditions.

Reply:
We use `ssize_t` in GATT for that actually, but (unrelated to this) we seem to misusing it, as `ssize_t` is only guaranteed to hold `[-1, SSIZE_MAX]`, but we use it with `BT_GATT_ERR` and an ATT error code, which certainly can be less than `-1`...

Reply:
@Thalley I think we need to look at the actual errors and consider what useful stuff the caller can do with that information. If you’re disconnected, then returning 0 is just the truth (you can t have any channels if you’re not connected). The NULL pointer case we can discuss, but that could even warrant an assert.

Reply:
I see your point.